### PR TITLE
Update new keyboard templates

### PR DESCRIPTION
**Pull request by joelspadin**
Updated the new keyboard templates to use Studio features such as physical layout definitions. Also updated the links to documentation for pages that have moved.

### CODE REVIEW

File: zmk/templates/common/layouts.dtsi [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// See https://zmk.dev/docs/development/hardware-integration/physical-layouts
+// for how to configure this.
+
+#include <physical_layouts.dtsi>
+
+/ {
+    default_layout: default_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Default Layout";
+        transform = <&default_layout>;
```
Comment (caksoylar) [inline zmk/templates/common/layouts.dtsi:10]:
```suggestion
        transform = <&default_transform>;
```

File: zmk/templates/board/nrf52840/common_outer.dtsi [hunk 2]
```diff
@@ -14,46 +14,50 @@
...
-        compatible = "zephyr,cdc-acm-uart";
-    };
+zephyr_udc0: &usbd {
+    status = "okay";
 };
 
 &flash0 {
     // Adjust this flash map as is necessary for your board. For more information, see
-    // https://docs.zephyrproject.org/latest/reference/storage/flash_map/flash_map.html
+    // https://docs.zephyrproject.org/3.6.0/services/storage/flash_map/flash_map.html#relationship-with-devicetree
```
Comment (caksoylar) [inline zmk/templates/board/nrf52840/common_outer.dtsi:31]:
```suggestion
    // https://docs.zephyrproject.org/3.5.0/services/storage/flash_map/flash_map.html#relationship-with-devicetree
```
Seems like the same page, so linking for ZMK's version might be better.

File: zmk/templates/common/layouts.dtsi [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        kscan = <&kscan>;
+
+        // Edit this to define the positions and sizes of the keys
+        // https://zmk.dev/docs/development/hardware-integration/physical-layouts#optional-keys-property
+        keys  //                     w   h    x    y     rot    rx    ry
```
Comment (caksoylar) [inline zmk/templates/common/layouts.dtsi:15]: Is it worth commenting this out since it is technically optional? I can see it either way.

Comment (joelspadin) [reply zmk/templates/common/layouts.dtsi:15]: I think it's required in order to use Studio? I'm not sure how Studio behaves if this is not set.

I could add a comment saying you can remove it if not intending to support Studio though.

Comment (caksoylar) [reply zmk/templates/common/layouts.dtsi:15]: Yeah, that sounds good.

File: zmk/templates/common/layouts.dtsi (hunk 1 above)
Comment (joelspadin) [reply zmk/templates/common/layouts.dtsi]: Whoops. Fixed.

File: zmk/templates/board/nrf52840/common_outer.dtsi (hunk 2 above)
Comment (joelspadin) [reply zmk/templates/board/nrf52840/common_outer.dtsi]: Looks like I had accidentally used 3.6.0 for all the other links, so I just updated this to match. Switched everything back to 3.5.0.

File: zmk/templates/common/layouts.dtsi (hunk 3 above)
Comment (joelspadin) [reply zmk/templates/common/layouts.dtsi:16]: Updated. Does that look better to you?

File: zmk/templates/board/nrf52840/common_outer.dtsi [hunk 4]
```diff
@@ -14,46 +14,50 @@
...
 
-&usbd {
-    status = "okay";
+&uart0 {
+    compatible = "nordic,nrf-uarte";
+    current-speed = <115200>;
+    pinctrl-0 = <&uart0_default>;
+    pinctrl-1 = <&uart0_sleep>;
+    pinctrl-names = "default", "sleep";
+};
```
Comment (joelspadin) [inline zmk/templates/board/nrf52840/common_outer.dtsi:23]: This is present in all the current nRF52 boards in ZMK, but I'm actually not sure if this is needed for anything.

File: zmk/templates/common/layouts.dtsi (hunk 3 above)
Comment (caksoylar) [reply zmk/templates/common/layouts.dtsi:16]: Yes, thanks!

File: zmk/templates/board/nrf52840/common_outer.dtsi (hunk 4 above)
Comment (Nick-Munnich) [reply zmk/templates/board/nrf52840/common_outer.dtsi:23]: I'm not sure it is currently, but I think it's useful to leave in to make wired split in the future easier.

**Pull request merged**